lib/Fhp/Action/SendSEPADirectDebit.php

```diff
@@ -105,16 +135,36 @@ public function createRequest(BPD $bpd, ?UPD $upd)
 
         /** @var HKDMEv1|HKDSEv1 $hkdxe */
         $hkdxe = null;
-        switch ($hidxes->getVersion()) {
-            case 1:
-                $hkdxe = $useSingleDirectDebit ? HKDSEv1::createEmpty() : HKDMEv1::createEmpty();
-            break;
-            case 2:
-                $hkdxe = $useSingleDirectDebit ? HKDSEv2::createEmpty() : HKDMEv2::createEmpty();
-            break;
-            default:
-                throw new UnsupportedException('Unsupported HKDME or HKDSE version: ' . $hidxes->getVersion());
-        }
+		switch ($this->coreType) {
```

Comment:
Consider adding a new function to the `HIXXES` interface: `public function createRequestSegment(): HKXXE` that you override in every subclass of the parameter classes and then just call once here. That would get rid of the many nested `switch` statements and ternary expressions. (My IDE would complain about cyclomatic complexity otherwise ;-) )

Reply:
not sure I understand.

Reply:
After you push a commit with the formatting fixed, I can send a commit with what I meant.